src/net/sourceforge/kolmafia/session/PingManager.java

```diff
@@ -0,0 +1,61 @@
+package net.sourceforge.kolmafia.session;
+
+import java.util.ArrayList;
+import java.util.List;
+import net.sourceforge.kolmafia.RequestLogger;
+import net.sourceforge.kolmafia.request.PingRequest;
+
+public class PingManager {
+
+  public PingManager() {}
+
+  public static class PingTest {
+    private final List<Long> pings = new ArrayList<>();
+    private long total;
+
+    public PingTest() {
+      this.pings.clear();
```

Comment:
given pings is instantiated as empty and isn't static, this shouldn't do anything?

Reply:
Right.

Other than saving pings in that list, we don't currently even use the data. I had some nebulous idea that perhaps the GUI could display the actual values to the user, to let them see just how bad (or good) things are - like Irrat's script does.

I think the average ping time is the actual metric we'll want to use to checking "goodness" computationally.